### PR TITLE
Add Extra ModelLoader studio plugin

### PR DESCRIPTION
**Pull request by emilia-sokol-gs**
## Summary

Add extra model loader studio plugin to allow new model loader types to be added in more modularized fashion

## How did you test this change?

The change will be tested when extra model loader plugin will be included

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [x]  No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-studio/src/components/editor/edit-panel/ModelLoader.tsx [hunk 1]
```diff
@@ -187,25 +214,35 @@ export const ModelLoader = observer(() => {
...
-            className="btn--dark model-loader__header__load-btn"
-            onClick={loadModel}
-            disabled={modelLoaderState.isLoadingModel}
-            tabIndex={-1}
-            title="Load model"
-          >
-            Load
-          </button>
-        </div>
+        {!modelLoaderState.currentExtraNativeInputType && (
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/edit-panel/ModelLoader.tsx:217]: Might make sense to add this as part of the extension definition. So we could add a `hardReplaceOption?: boolean | undefined` property on `ExtraModelLoaderRendererConfiguration` to see if we render this or not.

File: packages/legend-studio/src/stores/LegendStudioPlugin.ts [hunk 2]
```diff
@@ -52,6 +53,12 @@ export type ClassPreviewRenderer = (
   _class: Class,
 ) => React.ReactNode | undefined;
 
+export type ExtraModelLoaderRendererConfiguration = {
+  key: string;
+  defaultModelText: string;
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/LegendStudioPlugin.ts:58]: `ExtraModelLoaderRendererConfiguration` -> `ExtraModelLoaderExtensionsConfiguration`
We probably don't need `defaultModelText` as that is specific to external and native types. 
Also it probably make sense to add a `label?: string` property to this as we may not always want to use the key in the render option.

File: packages/legend-studio/src/components/editor/edit-panel/ModelLoader.tsx [hunk 3]
```diff
@@ -187,25 +214,35 @@ export const ModelLoader = observer(() => {
...
+          <div className="model-loader__header__action">
+            <button
+              className="btn--dark model-loader__header__load-btn"
+              onClick={loadModel}
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/edit-panel/ModelLoader.tsx:221]: I think we also need an extension when user clicks `load`. Your extension will handle that click as load/import your model to studio.

**Comment by MauricioUyaguari** [comment]
Lets also add a changset

**Comment by akphi** [comment]
@emilia-sokol-gs I think the pipelines are not running for you. I don't know the exact reasons but try to force push again to see if it was a transient issue, otherwise, the only explanation I can think of is that since you haven't worked on `finos/legend-studio` in a while, `Github` auto-disable github actions on your fork

**Comment by emilia-sokol-gs** [comment]
@akphi I pushed those changes again from another branch (I was using master so far) and this fixed this issue
@MauricioUyaguari I applied your feedback here: https://github.com/finos/legend-studio/pull/980 so I will close this PR

**Comment by emilia-sokol-gs** [comment]
PR is currently moved here: https://github.com/finos/legend-studio/pull/980

**Pull request closed without merging**